symengine/fields.h

```diff
@@ -229,7 +258,8 @@ class GaloisFieldDict
         GaloisFieldDict o(*this);
         for (auto &a : o.dict_) {
             a *= -1;
-            a += modulo_;
+            if (a != 0_z)
```

Comment:
Before this change, `[0, 0, 1]` was negating to `[11, 11, 10]` in finite field(11)